File: Python/WwiseSoundbankXMLParser.py
// ©2023 JDSherbert

import xml.etree.ElementTree as ET

def get_events(soundbank_xml_path):
    event_map = {}

    tree = ET.parse(soundbank_xml_path)
    root = tree.getroot()

    events = root.find("./SoundBank/Events")
    for event in events.findall("Event"):
        event_id = event.find("ID").text
        event_name = event.find("Name").text

        if event_id and event_name:
            event_map[event_id] = event_name

    return event_map

def get_rtpcs(soundbank_xml_path):
    rtpc_map = {}

    tree = ET.parse(soundbank_xml_path)
    root = tree.getroot()

    rtpcs = root.find("./SoundBank/RealTimeParameters")
    for rtpc in rtpcs.findall("RealTimeParameter"):
        rtpc_id = rtpc.find("ID").text
        rtpc_name = rtpc.find("Name").text

        if rtpc_id and rtpc_name:
            rtpc_map[rtpc_id] = rtpc_name

    return rtpc_map

def get_switches(soundbank_xml_path):
    switch_map = {}

    tree = ET.parse(soundbank_xml_path)
    root = tree.getroot()

    switches = root.find("./SoundBank/Switches")
    for switch in switches.findall("Switch"):
        switch_id = switch.find("ID").text
        switch_name = switch.find("Name").text

        if switch_id and switch_name:
            switch_map[switch_id] = switch_name

    return switch_map

def get_states(soundbank_xml_path):
    state_map = {}

    tree = ET.parse(soundbank_xml_path)
    root = tree.getroot()

    states = root.find("./SoundBank/States")
    for state in states.findall("State"):
        state_id = state.find("ID").text
        state_name = state.find("Name").text

        if state_id and state_name:
            state_map[state_id] = state_name

    return state_map

def get_auxbuses(soundbank_xml_path):
    auxbus_map = {}

    tree = ET.parse(soundbank_xml_path)
    root = tree.getroot()

    auxbuses = root.find("./SoundBank/AuxBus")
    for auxbus in auxbuses.findall("AuxBus"):
        auxbus_id = auxbus.find("ID").text
        auxbus_name = auxbus.find("Name").text

        if auxbus_id and auxbus_name:
            auxbus_map[auxbus_id] = auxbus_name

    return auxbus_map

File: Java/WwiseSoundbankXMLParser.java
// ©2023 JDSherbert

import org.w3c.dom.Document;
import org.w3c.dom.Element;
import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;
import java.io.File;
import java.util.HashMap;
import java.util.Map;

public class WwiseSoundbankParser {

    public static Map<String, String> getEventIDAndName(String soundbankXmlPath) {
        Map<String, String> eventMap = new HashMap<>();

        try {
            File soundbankXmlFile = new File(soundbankXmlPath);
            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
            DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
            Document doc = dBuilder.parse(soundbankXmlFile);

            doc.getDocumentElement().normalize();
            NodeList eventList = doc.getElementsByTagName("Event");

            for (int i = 0; i < eventList.getLength(); i++) {
                Node eventNode = eventList.item(i);
                if (eventNode.getNodeType() == Node.ELEMENT_NODE) {
                    Element eventElement = (Element) eventNode;
                    String eventID = eventElement.getElementsByTagName("ID").item(0).getTextContent();
                    String eventName = eventElement.getElementsByTagName("Name").item(0).getTextContent();

                    if (eventID != null && eventName != null) {
                        eventMap.put(eventID, eventName);
                    }
                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

        return eventMap;
    }

    public static Map<String, String> getRTPCIDAndName(String soundbankXmlPath) {
        Map<String, String> rtpcMap = new HashMap<>();

        try {
            File soundbankXmlFile = new File(soundbankXmlPath);
            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
            DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
            Document doc = dBuilder.parse(soundbankXmlFile);

            doc.getDocumentElement().normalize();
            NodeList rtpcList = doc.getElementsByTagName("RealTimeParameter");

            for (int i = 0; i < rtpcList.getLength(); i++) {
                Node rtpcNode = rtpcList.item(i);
                if (rtpcNode.getNodeType() == Node.ELEMENT_NODE) {
                    Element rtpcElement = (Element) rtpcNode;
                    String rtpcID = rtpcElement.getElementsByTagName("ID").item(0).getTextContent();
                    String rtpcName = rtpcElement.getElementsByTagName("Name").item(0).getTextContent();

                    if (rtpcID != null && rtpcName != null) {
                        rtpcMap.put(rtpcID, rtpcName);
                    }
                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

        return rtpcMap;
    }

    public static Map<String, String> getSwitchIDAndName(String soundbankXmlPath) {
        Map<String, String> switchMap = new HashMap<>();

        try {
            File soundbankXmlFile = new File(soundbankXmlPath);
            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
            DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
            Document doc = dBuilder.parse(soundbankXmlFile);

            doc.getDocumentElement().normalize();
            NodeList switchList = doc.getElementsByTagName("Switch");

            for (int i = 0; i < switchList.getLength(); i++) {
                Node switchNode = switchList.item(i);
                if (switchNode.getNodeType() == Node.ELEMENT_NODE) {
                    Element switchElement = (Element) switchNode;
                    String switchID = switchElement.getElementsByTagName("ID").item(0).getTextContent();
                    String switchName = switchElement.getElementsByTagName("Name").item(0).getTextContent();

                    if (switchID != null && switchName != null) {
                        switchMap.put(switchID, switchName);
                    }
                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

        return switchMap;
    }

    public static Map<String, String> getStateIDAndName(String soundbankXmlPath) {
        Map<String, String> stateMap = new HashMap<>();

        try {
            File soundbankXmlFile = new File(soundbankXmlPath);
            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
            DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
            Document doc = dBuilder.parse(soundbankXmlFile);

            doc.getDocumentElement().normalize();
            NodeList stateList = doc.getElementsByTagName("State");

            for (int i = 0; i < stateList.getLength(); i++) {
                Node stateNode = stateList.item(i);
                if (stateNode.getNodeType() == Node.ELEMENT_NODE) {
                    Element stateElement = (Element) stateNode;
                    String stateID = stateElement.getElementsByTagName("ID").item(0).getTextContent();
                    String stateName = stateElement.getElementsByTagName("Name").item(0).getTextContent();

                    if (stateID != null && stateName != null) {
                        stateMap.put(stateID, stateName);
                    }
                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

        return stateMap;
    }

    public static Map<String, String> getAuxBusIDAndName(String soundbankXmlPath) {
        Map<String, String> auxBusMap = new HashMap<>();

        try {
            File soundbankXmlFile = new File(soundbankXmlPath);
            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
            DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
            Document doc = dBuilder.parse(soundbankXmlFile);

            doc.getDocumentElement().normalize();
            NodeList auxBusList = doc.getElementsByTagName("AuxBus");

            for (int i = 0; i < auxBusList.getLength(); i++) {
                Node auxBusNode = auxBusList.item(i);
                if (auxBusNode.getNodeType() == Node.ELEMENT_NODE) {
                    Element auxBusElement = (Element) auxBusNode;
                    String auxBusID = auxBusElement.getElementsByTagName("ID").item(0).getTextContent();
                    String auxBusName = auxBusElement.getElementsByTagName("Name").item(0).getTextContent();

                    if (auxBusID != null && auxBusName != null) {
                        auxBusMap.put(auxBusID, auxBusName);
                    }
                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

        return auxBusMap;
    }
